clients/fides-js/src/lib/tcf.ts

```diff
@@ -119,37 +153,73 @@ export const generateTcString = async ({
 };
 
 /**
- * Call tcf() to configure Fides with tcf support (if tcf is enabled).
+ * Extract just the TC string from a FidesEvent. This will also remove parts of the
+ * TC string that we do not want to surface with our CMP API events, such as
+ * `vendors_disclosed` and our own AC string addition.
  */
-export const tcf = () => {
+const fidesEventToTcString = (event: FidesEvent) => {
+  const { fides_tc_string: cookieString } = event.detail;
+  if (cookieString) {
+    // Remove the AC portion which is separated by FIDES_SEPARATOR
+    const [tcString] = cookieString.split(FIDES_SEPARATOR);
+    // We only want to return the first part of the tcString, which is separated by '.'
+    // This means Publisher TC is not sent either, which is okay for now since we do not set it.
+    // However, if we do one day set it, we would have to decode the string and encode it again
+    // without vendor_consents
+    return tcString.split(".")[0];
+  }
+  return cookieString;
+};
+
+/**
+ * Initializes the CMP API, including setting up listeners on FidesEvents to update
+ * the CMP API accordingly.
+ */
+export const initializeCmpApi = () => {
   makeStub();
   const isServiceSpecific = true; // TODO: determine this from the backend?
-  const cmpApi = new CmpApi(CMP_ID, CMP_VERSION, isServiceSpecific);
+  const cmpApi = new CmpApi(CMP_ID, CMP_VERSION, isServiceSpecific, {
+    // Add custom command to support adding `addtlConsent` per AC spec
+    getTCData: (next, tcData: TCData, status) => {
```

Comment:
adapted from the docs: https://github.com/InteractiveAdvertisingBureau/iabtcf-es/tree/master/modules/cmpapi#built-in-and-custom-commands